src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py

```diff
@@ -1065,11 +1065,28 @@ def test_evaluate(case_setup):
         writer.finished_ok = True
 
 
-def test_exception_details(case_setup):
-    with case_setup.test_file('_debugger_case_exceptions.py') as writer:
+@pytest.mark.parametrize('max_frames', ['default', 'all', 10])  # -1 = default, 0 = all, 10 = 10 frames
+def test_exception_details(case_setup, max_frames):
+    with case_setup.test_file('_debugger_case_large_exception_stack.py') as writer:
         json_facade = JsonFacade(writer)
 
         writer.write_set_protocol('http_json')
+        if max_frames == 'all':
+            json_facade.write_launch(maxExcpetionStackFrames=0)
+            # trace back compresses repeated text
+            min_expected_lines = 10
+            max_expected_lines = 1000
```

Comment:
Given that the difference is so large, it doesn't seem like this test will fail when it should if something goes awry. 

Maybe you can change the code to have 2 methods recursing (that way I think Python won't compress it as the same function won't repeat over and over in the traceback) -- likewise for the test in `ptvsd`.